src/audiences/ui/components/rule.js

```diff
@@ -0,0 +1,248 @@
+import React, { Fragment, useState, useRef, useEffect } from 'react';
+import styled from 'styled-components';
+
+import SelectOperator from './select-operator';
+
+const { Button } = wp.components;
+const { useSelect } = wp.data;
+const { sprintf, __ } = wp.i18n;
+
+const StyledRule = styled.div`
+	margin: 0 0 15px;
+	display: flex;
+	flex-wrap: wrap;
+	align-items: center;
+
+	select, input {
+		flex: 1;
+	}
+
+	&& > * + * {
+		margin-left: 5px;
+	}
+
+	.audience-editor__rule-operator,
+	button {
+		flex 0;
+	}
+
+	p.description {
+		width: 100%;
+		margin: 8px 0 0;
+	}
+`;
+
+const ClearableInput = styled.div`
+	flex: 1;
+	input {
+		width: 100%;
+	}
+	.components-button {
+		text-decoration: none;
+		margin-left: -1.5em;
+
+		&:hover {
+			color: #d94f4f;
+		}
+	}
+`;
+
+const RuleInput = props => {
+	const {
+		disabled,
+		currentField,
+		name,
+		operator,
+		value,
+		onChange,
+	} = props;
+
+	// Get the currently available values for the field.
+	const data = ( currentField.data && currentField.data.map( datum => datum.value ) ) || [];
+
+	// Get input refs to manage focus.
+	const inputEl = useRef( null );
+	const dropdownRef = useRef( null );
+
+	// Default to display of the dropdown if the value is in existing data or empty.
+	const [ showDropdown, setShowDropdown ] = useState( value === '' || data.indexOf( value ) >= 0 );
+	useEffect( () => {
+		if ( showDropdown ) {
+			dropdownRef && dropdownRef.current && dropdownRef.current.focus();
+		} else {
+			inputEl && inputEl.current && inputEl.current.focus();
+		}
+	}, [ showDropdown ] );
+
+
+	switch ( currentField.type ) {
+		case 'number':
+			return (
+				<input
+					className="regular-text"
+					disabled={ disabled }
+					name={ name }
+					placeholder={ `${ sprintf(
+						__( 'Avg: %d, Lowest: %d, Highest: %d', 'altis-analytics' ),
+						Number( currentField.stats.avg ).toFixed( 2 ),
+						Number( currentField.stats.min ).toFixed( 2 ),
+						Number( currentField.stats.max ).toFixed( 2 )
+					) }` }
+					type="number"
+					value={ value }
+					onChange={ onChange }
+				/>
+			);
+
+		case 'string':
+		default:
+			switch ( operator ) {
+				case '=':
+				case '!=':
+				case '*=':
+				case '!*':
+				case '^=':
+					return (
+						<Fragment>
+							{ ! showDropdown && (
+								<ClearableInput>
+									<input
+										ref={ inputEl }
+										className="regular-text"
+										disabled={ disabled }
+										name={ name }
+										type="text"
+										value={ value }
+										onChange={ onChange }
+										list={ `audience-editor__rule__data_${ name }` }
```

Comment:
Is `list` actually even used now or is that a holdover from datalist?
```suggestion
										list={ `audience-editor__rule__data_${ name }` }
										name={ name }
										type="text"
										value={ value }
										onChange={ onChange }
```